route4me-csharp-sdk/Route4MeSDKLibrary/Route4MeManagerV5.cs

```diff
@@ -1960,153 +1910,554 @@ public VehicleTrackResponse GetVehicleTrack(string vehicleId, out ResultResponse
-                new VehicleParameters(),
-                R4MEInfrastructureSettingsV5.VehicleProfiles + "/" + vehicleProfileParams.VehicleProfileId,
-                HttpMethodType.Get,
+            var updateBodyJsonString = R4MeUtils.SerializeObjectToJson(vehicleParams, true);
+
+            var content = new StringContent(updateBodyJsonString, Encoding.UTF8, "application/json");
+
+            var genParams = new RouteParametersQuery();
+
+            var result = GetJsonObjectFromAPI<Vehicle>(
+                genParams,
+                R4MEInfrastructureSettingsV5.Vehicles + "/" + vehicleParams.VehicleId,
+                HttpMethodType.Patch,
+                content,
                 out resultResponse);
 
             return result;
         }
 
         /// <summary>
-        ///     Get vehicle by license plate.
+        ///     Update a vehicle
         /// </summary>
-        /// <param name="vehicleParams">Vehicle parameter containing vehicle license plate</param>
-        /// <param name="resultResponse">Failing response</param>
-        /// <returns>Vehicle</returns>
-        public VehicleResponse GetVehicleByLicensePlate(VehicleParameters vehicleParams,
-            out ResultResponse resultResponse)
+        /// <param name="vehicleParams">Vehicle body parameters</param>
+        /// <returns>Updated vehicle </returns>
+        public Task<Tuple<Vehicle, ResultResponse>> UpdateVehicleAsync(Vehicle vehicleParams)
         {
-            if ((vehicleParams?.VehicleLicensePlate?.Length ?? 0) < 1)
+            if (vehicleParams == null || (vehicleParams.VehicleId?.Length ?? 0) != 32)
             {
-                resultResponse = new ResultResponse
+                return Task.FromResult(new Tuple<Vehicle, ResultResponse>(null, new ResultResponse
                 {
                     Status = false,
                     Messages = new Dictionary<string, string[]>
                     {
-                        {"Error", new[] {"The vehicle license plate is not specified"}}
+                        {"Error", new[] {"The vehicle ID is not specified"}}
                     }
-                };
-
-                return null;
+                }));
             }
 
-            var result = GetJsonObjectFromAPI<VehicleResponse>(
-                vehicleParams,
-                R4MEInfrastructureSettingsV5.VehicleLicense,
+            var updateBodyJsonString = R4MeUtils.SerializeObjectToJson(vehicleParams, false);
+
+            var content = new StringContent(updateBodyJsonString, Encoding.UTF8, "application/json");
+
+            var genParams = new RouteParametersQuery();
+
+            var result = GetJsonObjectFromAPIAsync<Vehicle>(
+                genParams,
+                R4MEInfrastructureSettingsV5.Vehicles + "/" + vehicleParams.VehicleId,
+                HttpMethodType.Patch,
+                content,
+                false,
+                false);
+
+            return Task.FromResult(
```

Comment:
you can just return result.

Reply:
See the comment above for similar question.